Sources/Vehicle/Extensions/HAEntityExtension.swift

```diff
@@ -0,0 +1,215 @@
+                    } else {
+                        image = MaterialDesignIcons.gestureTapButtonIcon
+                    }
+                case "cover":
+                    image = getCoverIcon()
+                case "input_boolean":
+                    if (!entityId.hasSuffix(".ha_ios_placeholder")) {
+                        if (compareState == "on") {
+                            image = MaterialDesignIcons.checkCircleOutlineIcon
+                        } else {
+                            image = MaterialDesignIcons.closeCircleOutlineIcon
+                        }
+                    } else {
+                        image = MaterialDesignIcons.toggleSwitchOutlineIcon
+                    }
+                case "input_button":
+                    image = MaterialDesignIcons.gestureTapButtonIcon
+                case "light":
+                    image = MaterialDesignIcons.lightbulbIcon
+                case "lock":
+                    switch (compareState) {
+                    case "unlocked":
+                        image = MaterialDesignIcons.lockOpenIcon
+                    case "jammed":
+                        image = MaterialDesignIcons.lockAlertIcon
+                    case "locking", "unlocking":
+                        image = MaterialDesignIcons.lockClockIcon
+                    default:
+                        image = MaterialDesignIcons.lockIcon
+                    }
+                case "person":
+                    image = MaterialDesignIcons.accountIcon
+                case "scene":
+                    image = MaterialDesignIcons.paletteOutlineIcon
+                case "script":
+                    image = MaterialDesignIcons.scriptTextOutlineIcon
+                case "sensor":
+                    image = MaterialDesignIcons.eyeIcon
+                case "switch":
+                if (!entityId.hasSuffix(".ha_ios_placeholder")) {
+                    let deviceClass = attributes.dictionary["device_class"] as? String
+                    switch(deviceClass) {
+                    case "outlet":
+                        image = compareState == "on" ? MaterialDesignIcons.powerPlugIcon : MaterialDesignIcons.powerPlugOffIcon
+                    case "switch":
+                        image = compareState == "on" ? MaterialDesignIcons.toggleSwitchIcon : MaterialDesignIcons.toggleSwitchOffIcon
+                    default:
+                        image = MaterialDesignIcons.flashIcon
+                    }
+                } else {
+                    image = MaterialDesignIcons.lightSwitchIcon
+                }
+            case "zone":
+                image = MaterialDesignIcons.mapMarkerRadiusIcon
+            default:
+                image = MaterialDesignIcons.bookmarkIcon
+            }
+        }
+        let iconImage = image.image(ofSize: size, color: nil)
+        iconImage.imageAsset?.register(image.image(ofSize: size, color: darkColor), with: .init(userInterfaceStyle: .dark))
```

Comment:
does the CarPlay UI not support tinting icons?

Reply:
afaik, it does not :/

Reply:
@ItzSwirlz is right, tinting icons are not supported in CarPlay.

Reply:
![Simulator Screenshot - iPhone 15 Pro - 2024-01-04 at 13 49 22](https://github.com/home-assistant/iOS/assets/5808343/47bd5365-4789-42c4-9472-10433254c85d)

Not exactly "tinting" but it is possible to change the color, check `Domain` and then the property `icon`